pycloudlib/azure/security_types.py

```diff
@@ -0,0 +1,136 @@
+
+
+class AzureStandardSecurityType(SecurityType):
+    """Azure Standard Security Type."""
+
+    name = "Standard"
+
+    def configure_vm_params(
+        self, vm_params: Dict[str, Any], **kwargs
+    ) -> Dict[str, Any]:
+        """Configure the vm_params to be passed to Azure.
+
+        Args:
+            vm_params: dict, The parameters passed to Azure
+            **kwargs: dictionary of security type parameters
+        """
+        return vm_params
+
+
+class AzureTrustedLaunchSecurityType(SecurityType):
+    """Azure Trusted Launch Security Type."""
+
+    name = "TrustedLaunch"
+
+    def configure_vm_params(
+        self, vm_params: Dict[str, Any], **kwargs
+    ) -> Dict[str, Any]:
+        """Configure the vm_params to be passed to Azure.
+
+        Args:
+            vm_params: dict, The parameters passed to Azure
+            **kwargs: dictionary of security type parameters
+        """
+        param_update = {
+            "security_profile": {
+                "security_type": self.name,
+                "uefi_settings": {
+                    "secure_boot_enabled": True,
+                    "v_tpm_enabled": True,
+                },
+            }
+        }
+        util.update_nested(vm_params, param_update)
+        return vm_params
+
+
+class AzureConfidentialVMSecurityType(SecurityType):
+    """Azure Confidential VM Security Type."""
+
+    name = "ConfidentialVM"
+
+    def configure_vm_params(
+        self, vm_params: Dict[str, Any], **kwargs
+    ) -> Dict[str, Any]:
+        """Configure the vm_params to be passed to Azure.
+
+        Args:
+            vm_params: dict, The parameters passed to Azure
+            **kwargs: dictionary of security type parameters
+        """
+        os_disk_enc: AzureCVMOSDiskEncryption = kwargs.get(
+            "os_disk_encryption",
+            AzureCVMOSDiskEncryption.DISK_WITH_VM_GUEST_STATE,
+        )
+        param_update = {
+            "security_profile": {
+                "security_type": self.name,
+                "uefi_settings": {
+                    "secure_boot_enabled": True,
+                    "v_tpm_enabled": True,
+                },
+            },
+            "storage_profile": {
+                "os_disk": {
+                    "create_option": "FromImage",
+                    "delete_option": "Delete",
+                    "managed_disk": {
+                        "storage_account_type": "Standard_LRS",
+                        "security_profile": {
+                            "security_encryption_type": os_disk_enc.value,
+                        },
+                    },
+                }
+            },
+        }
+        util.update_nested(vm_params, param_update)
+        return vm_params
+
+    def is_correct_vm_size(self, vm_size: str) -> bool:
+        """Check if the vm size is supported by security type.
+
+        Args:
+            vm_size: string, The
+        """
+        size = re.search(r"[DE]C\d+a[d]*s_v5", vm_size)
```

Comment:
Could we add some example / unit test covering / showcasing this ?

Reply:
Added.

Reply:
We will not need this since we do not check the vm size.